lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/stats/CubePercentile.java

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.futuredata.macrobase.analysis.classify.stats;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+
+import java.util.*;
+
+/**
+ * Computes percentiles when given an array of metrics and the count of the
+ * number of times each occurs. Useful for computing percentiles on cubed data.
+ */
+public class CubePercentile {
+    private double[] counts;
+    private double[] metrics;
+
+    // Computed
+    private int numRawMetrics = 0;
+    private Multiset<Double> metricCounts;
+    private List<Double> sortedMetrics;
+
+    public CubePercentile(double[] counts, double[] metrics) {
+        this.counts = counts;
+        this.metrics = metrics;
+        computeCounts();
+    }
+
+    public double evaluate(double percentile) {
+        if (percentile >= 50.0) {
+            int numToPass = (int)((100.0 - percentile) / 100.0 * numRawMetrics);
+            int numPassed = 0;
+            for (int i = sortedMetrics.size() - 1; i >= 0; i--) {
+                numPassed += metricCounts.count(sortedMetrics.get(i));
```

Comment:
If you precalculated and cached of the running sum of the counts you probably wouldn't need to add in the loop. Could just do straight comparisons instead or even do binary search.